lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -838,36 +838,67 @@ class ADContext {
   SILPassManager &getPassManager() const { return passManager; }
   Lowering::TypeConverter &getTypeConverter() { return module.Types; }
 
-  SmallVectorImpl<DifferentiableFunctionInst *> &
-  getDifferentiableFunctionInsts() {
-    return differentiableFunctionInsts;
+  /// Returns true if there are `DifferentiableFunctionInst` to process.
+  bool hasDifferentiableFunctionInsts() const {
+    return !differentiableFunctionInsts.empty();
   }
 
-  SmallPtrSetImpl<DifferentiableFunctionInst *> &
-  getProcessedDifferentiableFunctionInsts() {
-    return processedDifferentiableFunctionInsts;
+  /// Pick and remove a `DifferentiableFunctionInst` for processing.
+  /// Returns nullptr if none are available for processing.
+  DifferentiableFunctionInst* pickAndRemoveDifferentiableFunctionInst() {
+    if (differentiableFunctionInsts.empty()) return nullptr;
+    auto *dfi = differentiableFunctionInsts.back();
```

Comment:
I believe you can replace these three lines with `differentiableFunctionInsts.pop_back_val()`, which pops and returns the back value.